File: 2041-lab-files/lab05/echon.pl
#!/usr/bin/perl -w
# check no. args
$error_args = "Usage: ./echon.pl <number of lines> <string>";
if (@ARGV != 2) {   # @ARGV in a scalar context
    die "$error_args";
}

# given integer n (arg0) and a string (arg1)
$n = $ARGV[0];
$line = $ARGV[1];
for ($i = 0; $i < $n; $i++) {
    print "$line\n";
}

File: 2041-lab-files/lab06/echon.py
#!/usr/bin/env python3
import sys
if len(sys.argv) != 3: # includes filename
    print("Usage: {} <number of lines> <string> at {}.".format(sys.argv[0], sys.argv[0]))
    sys.exit(1)
# up only by default
for i in range(0, int(sys.argv[1])):
    print(sys.argv[2])
